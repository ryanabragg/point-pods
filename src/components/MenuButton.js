import React from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';

import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import MenuList from '@material-ui/core/MenuList';
import MenuItem from '@material-ui/core/MenuItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import ClickAwayListener from '@material-ui/core/ClickAwayListener';
import Grow from '@material-ui/core/Grow';
import Paper from '@material-ui/core/Paper';
import Popper from '@material-ui/core/Popper';

const styles = theme => ({
  root: {
    display: 'flex',
  },
  paper: {
    marginRight: theme.spacing.unit * 2,
  },
});

class MenuButton extends React.Component {
  state = {
    open: false,
  };

  menuAnchor = null;

  handleToggle = () => this.setState(state => ({ open: !state.open }));

  handleClose = event => {
    if(this.menuAnchor.contains(event.target))
      return;
    this.setState({ open: false });
  };

  handleClick = (i) => (event) => {
    if(typeof this.props.menuItems[i].action === 'function')
      this.props.menuItems[i].action();
  }

  render() {
    const { classes, buttonType, buttonProps, buttonContent, menuItems } = this.props;
    const { open } = this.state;
    let TheButton;
    switch(buttonType) {
      default:
      case 'text': TheButton = Button; break;
      case 'icon': TheButton = IconButton; break;
    }

    return (
      <div className={classes.root}>
          <TheButton
            {...buttonProps}
            buttonRef={node => this.menuAnchor = node}
            aria-owns={open ? 'button-menu' : null}
            aria-haspopup='true'
            onClick={this.handleToggle}
          >
            {buttonContent}
          </TheButton>
          <Popper open={open} anchorEl={this.menuAnchor} transition disablePortal>
            {({ TransitionProps, placement }) => (
              <Grow
                {...TransitionProps}
                id='button-menu'
                style={{ transformOrigin: placement === 'bottom' ? 'center top' : 'center bottom' }}
              >
                <Paper>
                  <ClickAwayListener onClickAway={this.handleClose}>
                    <MenuList onClick={this.handleClose}>
                      {menuItems.map((item, i) => (
                        <MenuItem key={i}
                          onClick={this.handleClick(i)}
                        >
                          {item.icon && (
                            <ListItemIcon>
                              {item.icon}
                            </ListItemIcon>
                          )}
                          {item.icon
                            ? <ListItemText inset primary={item.label} />
                            : item.label
                          }
                        </MenuItem>
                      ))}
                    </MenuList>
                  </ClickAwayListener>
                </Paper>
              </Grow>
            )}
          </Popper>
      </div>
    );
  }
}

MenuButton.defaultProps = {
  buttonType: 'normal',
  buttonContent: 'button',
  buttonProps: {},
  menuItems: [],
}

MenuButton.propTypes = {
  classes: PropTypes.object.isRequired, // added by withStyles
  buttonType: PropTypes.oneOf(['normal', 'icon']),
  buttonContent: PropTypes.node,
  buttonProps: PropTypes.object,
  menuItems: PropTypes.array,
};

export default withStyles(styles)(MenuButton);
