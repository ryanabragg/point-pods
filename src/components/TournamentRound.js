import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';

import Typography from '@material-ui/core/Typography';
import Divider from '@material-ui/core/Divider';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardContent from '@material-ui/core/CardContent';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import Tooltip from '@material-ui/core/Tooltip';
import TextField from '@material-ui/core/TextField';
import IconButton from '@material-ui/core/IconButton';

import GroupWorkIcon from '@material-ui/icons/GroupWork';
import ArrowRightAltIcon from '@material-ui/icons/ArrowRightAlt';
import AddIcon from '@material-ui/icons/Add';
import DeleteIcon from '@material-ui/icons/Delete';
import RestoreFromTrashIcon from '@material-ui/icons/RestoreFromTrash';

import MenuButton from '../components/MenuButton';

const styles = theme => ({
  ctaStart: {
    position: 'absolute',
    top: 20,
    left: '50%',
    transform: 'translate(-50%, 0)',
  },
  unmargined: {
    margin: 0,
    padding: 0,
  },
  margined: {
    margin: theme.spacing.unit,
  },
  marginedLeft: {
    marginLeft: theme.spacing.unit,
  },
  menuButtonPaper: {
    maxHeight: theme.spacing.unit * 6 * 5.5,
    overflow: 'auto',
  },
  flex: {
    display: 'flex',
    flexWrap: 'wrap',
  },
  right: {
    marginLeft: 'auto',
  },
  error: {
    color: theme.palette.error.contrastText,
    backgroundColor: theme.palette.error.main,
    '&:hover': {
      backgroundColor: theme.palette.error.dark,
    },
  },
  scoreField: {
    maxWidth: 7 * theme.spacing.unit,
  },
  printPod: {
    width: '90%',
    marginTop: theme.spacing.unit,
    marginBottom: theme.spacing.unit,
    padding: 2 * theme.spacing.unit,
    borderTop: '1px dashed',
    pageBreakInside: 'avoid',
  },
  printPodEntry: {
    height: 3 * theme.spacing.unit,
    width: 8 * theme.spacing.unit,
    borderBottom: '1px solid',
  },
  hideOnPrint: {
    '@media print': {
      display: 'none',
    },
  },
  showOnPrint: {
    display: 'none',
    '@media print': {
      display: 'block',
    },
  },
});

class TournamentRounds extends Component {
  scores = [];

  componentWillUnmount() {
    if(this.timeout)
      clearTimeout(this.timeout);
    this.timeout = null;
    this.props.sync(false);
  }

  updateScores = () => {
    this.props.onUpdateScores(this.scores);
    this.scores = [];
  };

  handleScorePlayer = id => event => {
    const points = Number(event.target.value) || 0;
    this.props.sync(true);
    const index = this.scores.findIndex(score => score.id === id);
    if(index === -1)
      this.scores = [
        ...this.scores,
        { id: id, points: points }
      ];
    else
      this.scores = [
        ...this.scores.slice(0, index),
        { id: id, points: points },
        ...this.scores.slice(index + 1, this.scores.length)
      ];
    this.timeout = setTimeout(this.updateScores, 2000);
  };

  handleSetPlayerPod = (id, pod) => () => {
    this.props.onSetPlayerPod(id, pod);
  };

  handleRemovePlayer = (id, drop) => () => {
    if(drop)
      this.props.onRemovePlayer(id);
    else
      this.props.onReinstatePlayer(id);
  };

  render() {
    const {
      classes,
      players,
      playerSort,
      rounds,
      activeRound,
      done
    } = this.props;
    const currentRound = rounds === activeRound && !done;
    const newPlayers = players.filter(p => !p.hasOwnProperty(activeRound) && !p.dropped);
    const podPlayers = players.filter(p => p.hasOwnProperty(activeRound));
    const droppedPlayers = players.filter(p => !p.hasOwnProperty(activeRound) && p.dropped);
    const podCount = podPlayers.reduce((m, v) => Math.max(m, v[activeRound].pod), 0);
    let pods = [];
    if(activeRound) {
      for (var i = 1; i <= podCount; i++) {
        pods.push({
          pod: i,
          players: podPlayers.filter(p => p[activeRound].pod === i)
            .map(player => ({
              id: player.id,
              name: player.name,
              dropped: player.dropped,
              points: player[activeRound].points,
            }))
            .sort(playerSort)
        });
      }
    }
    return (
      <div>
        <div className={classes.hideOnPrint}>
          {newPlayers.length > 0 && (
            <Card className={classes.margined}>
              <CardHeader
                title={`${rounds ? '' : 'New '}Players`}
              />
              <Divider />
              <CardContent>
                <List>
                {newPlayers.map(({id, name}) => {
                  return (
                    <ListItem key={id} id={id} button>
                      {!currentRound ? null : (
                        <MenuButton
                          buttonType='icon'
                          buttonContent={
                            <Tooltip title={`Move ${name} to a different pod`}>
                              <GroupWorkIcon color='primary' />
                            </Tooltip>
                          }
                          menuItems={pods.map(d => ({
                            label: `Move to Pod ${d.pod} (${d.players.length} players)`,
                            icon: <ArrowRightAltIcon />,
                            action: this.handleSetPlayerPod(id, d.pod),
                          })).concat({
                            label: `Move to new pod (Pod ${pods.length + 1})`,
                            icon: <AddIcon />,
                            action: this.handleSetPlayerPod(id, pods.length + 1),
                          })}
                          paperStyle={this.props.classes.menuButtonPaper}
                        />
                      )}
                      <ListItemText primary={name} />
                      {!rounds || !currentRound ? null : (
                        <ListItemSecondaryAction>
                          <IconButton onClick={this.handleRemovePlayer(id, true)}>
                            <Tooltip title={`Remove ${name} from tournament`}>
                              <DeleteIcon color='primary' />
                            </Tooltip>
                          </IconButton>
                        </ListItemSecondaryAction>
                      )}
                    </ListItem>
                  );
                })}
                </List>
              </CardContent>
            </Card>
          )}
          {podPlayers.length > 0 &&
            pods.map(pod => (
              <Card key={pod.pod} className={classes.margined}>
                <CardHeader
                  title={`Pod ${pod.pod}`}
                />
                <Divider />
                <CardContent>
                  <List>
                  {pod.players.map(({id, name, dropped, points}) => {
                    let pointsSoFar = players.filter(p => p.id === id)[0].points;
                    return (
                      <ListItem key={id} id={id} button>
                        {!currentRound ? null : (
                          <MenuButton
                            buttonType='icon'
                            buttonContent={
                              <Tooltip title={`Move ${name} to a different pod`}>
                                <GroupWorkIcon className={classes.unmargined} color={dropped ? 'error' : 'primary'} />
                              </Tooltip>
                            }
                            menuItems={pods.filter(d => d.pod !== pod.pod).map(d => ({
                              label: `Move to Pod ${d.pod} (${d.players.length} players)`,
                              icon: <ArrowRightAltIcon color='primary' />,
                              action: this.handleSetPlayerPod(id, d.pod),
                            })).concat({
                              label: `Move to new pod (Pod ${pods.length + 1})`,
                              icon: <AddIcon />,
                              action: this.handleSetPlayerPod(id, pods.length + 1),
                            })}
                            paperStyle={this.props.classes.menuButtonPaper}
                          />
                        )}
                        <ListItemText primary={name} secondary={currentRound ? `${pointsSoFar} points` : null} />
                        <ListItemSecondaryAction>
                          {!currentRound ? points : (
                            <TextField
                              aria-label={`Points for player ${name}`}
                              margin='dense'
                              className={classes.scoreField}
                              required
                              type='number'
                              defaultValue={points}
                              onChange={this.handleScorePlayer(id)}
                            />
                          )}
                          {!currentRound ? null : (
                            <Tooltip title={dropped ? 'Reinstate Player' : 'Drop Player'}>
                              <IconButton
                                className={classes.marginedLeft}
                                onClick={this.handleRemovePlayer(id, !dropped)}
                              >
                                {dropped ? <RestoreFromTrashIcon color='error' /> : <DeleteIcon color='primary' />}
                              </IconButton>
                            </Tooltip>
                          )}
                        </ListItemSecondaryAction>
                      </ListItem>
                    );
                  })}
                  </List>
                </CardContent>
              </Card>
            ))
          }
          {droppedPlayers.length > 0 && (
            <Card className={classes.margined}>
              <CardHeader
                title={`Dropped Players`}
              />
              <Divider />
              <CardContent>
                <List>
                {droppedPlayers.map(({id, name}) => {
                  return (
                    <ListItem key={id} id={id} button>
                      {!currentRound ? null : (
                        <MenuButton
                          buttonType='icon'
                          buttonContent={
                            <Tooltip title={`Reinstate ${name} and add them to a Pod`}>
                              <GroupWorkIcon color='primary' />
                            </Tooltip>
                          }
                          menuItems={pods.map(d => ({
                            label: `Move to Pod ${d.pod} (${d.players.length} players)`,
                            icon: <ArrowRightAltIcon />,
                            action: this.handleSetPlayerPod(id, d.pod),
                          })).concat({
                            label: `Move to new pod (Pod ${pods.length + 1})`,
                            icon: <AddIcon />,
                            action: this.handleSetPlayerPod(id, pods.length),
                          })}
                          paperStyle={this.props.classes.menuButtonPaper}
                        />
                      )}
                      <ListItemText primary={name} />
                    </ListItem>
                  );
                })}
                </List>
              </CardContent>
            </Card>
          )}
        </div>
        <div className={classes.showOnPrint}>
          {pods.map(pod => (
            <div key={pod.pod} className={classes.printPod}>
              <Typography variant='headline' className={classes.flex}>
                {`Round ${activeRound}, Pod ${pod.pod}`}
                <span className={classes.right}>Points</span>
              </Typography>
              <List>
                {pod.players.map(({id, name}) => (
                  <ListItem key={id}>
                    <ListItemText primary={name} />
                    <ListItemSecondaryAction>
                      <div className={classes.printPodEntry} />
                    </ListItemSecondaryAction>
                  </ListItem>
                ))}
              </List>
            </div>
          ))}
          {!pods.length && (
            <div>
              <Typography variant='headline'>Participants</Typography>
              <List>
                {players.sort(playerSort).map(({ id, name }) => (
                  <ListItem key={id} dense className={classes.printList}>
                    <ListItemText primary={name} />
                  </ListItem>
                ))}
              </List>
            </div>
          )}
        </div>
      </div>
    );
  }
}

TournamentRounds.defaultProps = {
  players: [],
  playerSort: (a, b) => 0,
  rounds: 0,
  activeRound: 0,
  done: false,
  onSetPlayerPod: (id, pod) => null,
  onUpdateScores: scores => null,
  onRemovePlayer: id => null,
  onReinstatePlayer: id => null,
  sync: bool => null,
};

TournamentRounds.propTypes = {
  classes: PropTypes.object.isRequired, // added by withStyles
  players: PropTypes.array,
  playerSort: PropTypes.func,
  rounds: PropTypes.number,
  activeRound: PropTypes.number,
  done: PropTypes.bool,
  onSetPlayerPod: PropTypes.func,
  onUpdateScores: PropTypes.func,
  onRemovePlayer: PropTypes.func,
  onReinstatePlayer: PropTypes.func,
  sync: PropTypes.func,
};

export default withStyles(styles, { withTheme: true })(TournamentRounds);
