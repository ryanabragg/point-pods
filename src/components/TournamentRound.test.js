import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Card from '@material-ui/core/Card';
import ListItem from '@material-ui/core/ListItem';
import TextField from '@material-ui/core/TextField';
import IconButton from '@material-ui/core/IconButton';

import TournamentRound from './TournamentRound';
import MenuButton from './MenuButton';

import { testPlayers, testTournaments } from '../testData';

const playerSort = jest.fn((a, b) => {
  if(a.name < b.name)
    return -1;
  else if(a.name > b.name)
    return 1;
  return 0;
});

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TournamentRound />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('matches the prior snapshot', () => {
  const component = renderer.create(<TournamentRound />);
  expect(component.toJSON()).toMatchSnapshot();
  component.update(
    <TournamentRound
      players={testTournaments[0].players}
    />
  );
  expect(component.toJSON()).toMatchSnapshot();
  component.update(
    <TournamentRound
      players={testTournaments[0].players}
      playerSort={playerSort}
      rounds={testTournaments[0].rounds}
      activeRound={testTournaments[0].activeRound}
    />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

describe('actions', () => {
  test('scoring', (done) => {
    const spy = jest.fn();
    const HOC = shallow(
      <TournamentRound
        players={testTournaments[1].players}
        playerSort={playerSort}
        rounds={testTournaments[1].rounds}
        activeRound={2}
        onUpdateScores={spy}
      />
    );
    const component = HOC.dive();
    component.find('#mock3').find(TextField).simulate('change', {target: {value: '42'}});
    setTimeout(() => {
      component.update();
      try {
        expect(spy.mock.calls[0][0]).toEqual([{id: 'mock3', points: 42}]);
        done();
      } catch (error) {
        done.fail(error);
      }
    }, 3000);
  });

  test('drop a player', () => {
    const spy = jest.fn();
    const HOC = shallow(
      <TournamentRound
        players={testTournaments[1].players}
        playerSort={playerSort}
        rounds={testTournaments[1].rounds}
        activeRound={2}
        onRemovePlayer={spy}
      />
    );
    const component = HOC.dive();
    component.find('#mock4').find(IconButton).simulate('click');
    expect(spy.mock.calls[0][0]).toBe('mock4');
  });

  test('reinstate a player', () => {
    const spy = jest.fn();
    const HOC = shallow(
      <TournamentRound
        players={testTournaments[1].players}
        playerSort={playerSort}
        rounds={testTournaments[1].rounds}
        activeRound={2}
        onReinstatePlayer={spy}
      />
    );
    const component = HOC.dive();
    component.find('#mock1').find(IconButton).simulate('click');
    expect(spy.mock.calls[0][0]).toBe('mock1');
  });

  test('moving a player to a pod', () => {
    const spy = jest.fn();
    const HOC = shallow(
      <TournamentRound
        players={testTournaments[1].players}
        playerSort={playerSort}
        rounds={testTournaments[1].rounds}
        activeRound={2}
        onSetPlayerPod={spy}
      />
    );
    const component = HOC.dive();
    expect(component.find(Card).length).toBe(4);
    expect(component.find(ListItem).length).toBe(testTournaments[1].players.length);
    component.find(Card).forEach((node, i) => {
      let players = {
        0: 1,
        1: 2,
        2: 4,
        3: 1,
      };
      expect(node.find(ListItem).length).toBe(players[i]);
      let menuItems = node.find(MenuButton).at(0).prop('menuItems');
      if(i == 1) {
        expect(menuItems.length).toBe(2);
        expect(menuItems[0].label).toBe('Move to Pod 2 (4 players)');
        expect(menuItems[1].label).toBe('Move to new pod (Pod 3)');
      }
      else if(i == 2) {
        expect(menuItems.length).toBe(2);
        expect(menuItems[0].label).toBe('Move to Pod 1 (2 players)');
        expect(menuItems[1].label).toBe('Move to new pod (Pod 3)');
      }
      else {
        expect(menuItems.length).toBe(3);
        expect(menuItems[0].label).toBe('Move to Pod 1 (2 players)');
        expect(menuItems[1].label).toBe('Move to Pod 2 (4 players)');
        expect(menuItems[2].label).toBe('Move to new pod (Pod 3)');
      }
    });
    component.setProps({ activeRound: 1 });
    expect(component.find(Card).length).toBe(3);
    expect(component.find(ListItem).length).toBe(testTournaments[1].players.length);
    component.find(Card).forEach((node, i) => {
      let players = {
        0: 2,
        1: 3,
        2: 3,
        3: 0,
      };
      expect(node.find(ListItem).length).toBe(players[i]);
      expect(node.find(MenuButton).length).toBe(0);
      expect(node.find(IconButton).length).toBe(0);
    });
  });
});
