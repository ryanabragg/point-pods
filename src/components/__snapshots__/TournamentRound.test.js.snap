// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the prior snapshot 1`] = `<div />`;

exports[`matches the prior snapshot 2`] = `
<div>
  <div
    className="MuiPaper-root-18 MuiPaper-elevation1-21 MuiPaper-rounded-19 MuiCard-root-17 TournamentRounds-margined-11"
  >
    <div
      className="MuiCardHeader-root-45"
    >
      <div
        className="MuiCardHeader-content-48"
      >
        <span
          className="MuiTypography-root-51 MuiTypography-headline-56 MuiCardHeader-title-49"
        >
          New Players
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root-76"
    />
    <div
      className="MuiCardContent-root-80"
    >
      <ul
        className="MuiList-root-81 MuiList-padding-82"
      >
        <div
          className="MuiButtonBase-root-95 MuiListItem-root-85 MuiListItem-default-88 MuiListItem-gutters-92 MuiListItem-button-93"
          id="mock1"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiListItemText-root-98"
          >
            <span
              className="MuiTypography-root-51 MuiTypography-subheading-58 MuiListItemText-primary-101"
            >
              test1
            </span>
          </div>
          <span
            className="MuiTouchRipple-root-104"
          />
        </div>
        <div
          className="MuiButtonBase-root-95 MuiListItem-root-85 MuiListItem-default-88 MuiListItem-gutters-92 MuiListItem-button-93"
          id="mock2"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiListItemText-root-98"
          >
            <span
              className="MuiTypography-root-51 MuiTypography-subheading-58 MuiListItemText-primary-101"
            >
              test2
            </span>
          </div>
          <span
            className="MuiTouchRipple-root-104"
          />
        </div>
        <div
          className="MuiButtonBase-root-95 MuiListItem-root-85 MuiListItem-default-88 MuiListItem-gutters-92 MuiListItem-button-93"
          id="mock3"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiListItemText-root-98"
          >
            <span
              className="MuiTypography-root-51 MuiTypography-subheading-58 MuiListItemText-primary-101"
            >
              test3
            </span>
          </div>
          <span
            className="MuiTouchRipple-root-104"
          />
        </div>
      </ul>
    </div>
  </div>
  <div
    className="MuiPaper-root-18 MuiPaper-elevation1-21 MuiPaper-rounded-19 MuiCard-root-17 TournamentRounds-margined-11"
  >
    <div
      className="MuiCardHeader-root-45"
    >
      <div
        className="MuiCardHeader-content-48"
      >
        <span
          className="MuiTypography-root-51 MuiTypography-headline-56 MuiCardHeader-title-49"
        >
          Dropped Players
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root-76"
    />
    <div
      className="MuiCardContent-root-80"
    >
      <ul
        className="MuiList-root-81 MuiList-padding-82"
      >
        <li
          className="MuiListItem-container-86"
        >
          <div
            className="MuiButtonBase-root-95 MuiListItem-root-85 MuiListItem-default-88 MuiListItem-gutters-92 MuiListItem-button-93 MuiListItem-secondaryAction-94"
            id="mock0"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="MuiListItemText-root-98"
            >
              <span
                className="MuiTypography-root-51 MuiTypography-subheading-58 MuiListItemText-primary-101"
              >
                test0
              </span>
            </div>
            <span
              className="MuiTouchRipple-root-104"
            />
          </div>
          <div
            className="MuiListItemSecondaryAction-root-111"
          >
            <div>
              <button
                aria-haspopup="true"
                aria-owns={null}
                className="MuiButtonBase-root-95 MuiIconButton-root-113"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-118"
                >
                  <svg
                    aria-describedby={null}
                    aria-hidden="true"
                    className="MuiSvgIcon-root-126 MuiSvgIcon-colorPrimary-127"
                    color={undefined}
                    focusable="false"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    title="Reinstate test0 and add them to a Pod"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM8 17.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5zM9.5 8c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8zm6.5 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-104"
                />
              </button>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`matches the prior snapshot 3`] = `
<div>
  <div
    className="MuiPaper-root-18 MuiPaper-elevation1-21 MuiPaper-rounded-19 MuiCard-root-17 TournamentRounds-margined-11"
  >
    <div
      className="MuiCardHeader-root-45"
    >
      <div
        className="MuiCardHeader-content-48"
      >
        <span
          className="MuiTypography-root-51 MuiTypography-headline-56 MuiCardHeader-title-49"
        >
          Players
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root-76"
    />
    <div
      className="MuiCardContent-root-80"
    >
      <ul
        className="MuiList-root-81 MuiList-padding-82"
      >
        <div
          className="MuiButtonBase-root-95 MuiListItem-root-85 MuiListItem-default-88 MuiListItem-gutters-92 MuiListItem-button-93"
          id="mock1"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiListItemText-root-98"
          >
            <span
              className="MuiTypography-root-51 MuiTypography-subheading-58 MuiListItemText-primary-101"
            >
              test1
            </span>
          </div>
          <span
            className="MuiTouchRipple-root-104"
          />
        </div>
        <div
          className="MuiButtonBase-root-95 MuiListItem-root-85 MuiListItem-default-88 MuiListItem-gutters-92 MuiListItem-button-93"
          id="mock2"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiListItemText-root-98"
          >
            <span
              className="MuiTypography-root-51 MuiTypography-subheading-58 MuiListItemText-primary-101"
            >
              test2
            </span>
          </div>
          <span
            className="MuiTouchRipple-root-104"
          />
        </div>
        <div
          className="MuiButtonBase-root-95 MuiListItem-root-85 MuiListItem-default-88 MuiListItem-gutters-92 MuiListItem-button-93"
          id="mock3"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiListItemText-root-98"
          >
            <span
              className="MuiTypography-root-51 MuiTypography-subheading-58 MuiListItemText-primary-101"
            >
              test3
            </span>
          </div>
          <span
            className="MuiTouchRipple-root-104"
          />
        </div>
      </ul>
    </div>
  </div>
  <div
    className="MuiPaper-root-18 MuiPaper-elevation1-21 MuiPaper-rounded-19 MuiCard-root-17 TournamentRounds-margined-11"
  >
    <div
      className="MuiCardHeader-root-45"
    >
      <div
        className="MuiCardHeader-content-48"
      >
        <span
          className="MuiTypography-root-51 MuiTypography-headline-56 MuiCardHeader-title-49"
        >
          Dropped Players
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root-76"
    />
    <div
      className="MuiCardContent-root-80"
    >
      <ul
        className="MuiList-root-81 MuiList-padding-82"
      >
        <div
          className="MuiButtonBase-root-95 MuiListItem-root-85 MuiListItem-default-88 MuiListItem-gutters-92 MuiListItem-button-93"
          id="mock0"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiListItemText-root-98"
          >
            <span
              className="MuiTypography-root-51 MuiTypography-subheading-58 MuiListItemText-primary-101"
            >
              test0
            </span>
          </div>
          <span
            className="MuiTouchRipple-root-104"
          />
        </div>
      </ul>
    </div>
  </div>
</div>
`;
